[ews-build.webkit.org] Handle empty responses from GraphQL in safe-merge-queue

JonWBedard opened the pull request:
#### 04983089aca814631458eb7a2d20635f6b40cb3b
<pre>
[ews-build.webkit.org] Handle empty responses from GraphQL in safe-merge-queue
<a href="https://bugs.webkit.org/show_bug.cgi?id=281908">https://bugs.webkit.org/show_bug.cgi?id=281908</a>
<a href="https://rdar.apple.com/138403813">rdar://138403813</a>

Reviewed by NOBODY (OOPS!).

* Tools/CISupport/ews-build/steps.py:
(GitHubMixin.get_number_of_prs_with_label): Handle empty GraphQL response.
(RetrievePRDataFromLabel.getAllPRData): Ditto.
* Tools/CISupport/ews-build/steps_unittest.py: Add tests with empty GraphQL responses.
</pre><!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/04983089aca814631458eb7a2d20635f6b40cb3b

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/73081 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/52510 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/25889 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/77296 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/24319 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/75196 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/60315 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/292 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/57442 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/15923 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/76148 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/47451 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/62895 "Passed tests") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/41/builds/37859 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/44090 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/20365 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/22648 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/65946 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/20720 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/78960 "Built successfully") | 
| | [  ~~🛠 vision~~](https://ews-build.webkit.org/#/builders/128/builds/417 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/9 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/65877 "Passed tests") | 
| [✅ 🧪 services](https://ews-build.webkit.org/#/builders/28/builds/72757 "Passed tests") | [  ~~🛠 vision-sim~~](https://ews-build.webkit.org/#/builders/121/builds/413 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/62897 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/65154 "Passed tests") | 
| | [⏳ 🧪 vision-wk2 ](https://ews-build.webkit.org/#/builders/visionOS-2-Simulator-WK2-Tests-EWS "Waiting in queue, processing has not started yet") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/7145 "Passed tests") | | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/127/builds/356 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/125/builds/388 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/129/builds/373 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/124/builds/387 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

briannafan commented on Tools/CISupport/ews-build/steps.py:
```diff
@@ -263,7 +263,7 @@ def get_number_of_prs_with_label(self, label):
         response = yield self.query_graph_ql(query)
         if response:
-            num_prs = response['data']['repository']['pullRequests']['totalCount']
+            num_prs = (((response.get('data') or {}).get('repository') or {}).get('pullRequests') or {}).get('totalCount') or 0
```
Any of these fields being null would imply not retrieving the actual results, so there could be n PRs in safe-merge-queue but it would look like there's 0. I think any of these being null should result in an error instead of defaulting to 0.

JonWBedard committed:
[ews-build.webkit.org] Handle empty responses from GraphQL in safe-merge-queue
https://bugs.webkit.org/show_bug.cgi?id=281908
rdar://138403813

Reviewed by NOBODY (OOPS!).

* Tools/CISupport/ews-build/steps.py:
(GitHubMixin.get_number_of_prs_with_label): Handle empty GraphQL response.
(RetrievePRDataFromLabel.getAllPRData): Ditto.
* Tools/CISupport/ews-build/steps_unittest.py: Add tests with empty GraphQL responses.
